src/coreclr/jit/emitarm64.cpp

```diff
@@ -8795,75 +8835,81 @@ void emitter::emitIns_R_R_R(
+            assert(sopt == INS_SCALABLE_OPTS_NONE);
             fmt = IF_SVE_CL_3A;
             break;
 
         case INS_sve_clasta:
         case INS_sve_clastb:
+            assert(insOptsScalable(opt));
             assert(isLowPredicateRegister(reg2));
             assert(isVectorRegister(reg3));
-            if (insOptsScalableSimple(opt))
+            if (isGeneralRegister(reg1))
             {
-                assert(isVectorRegister(reg1));
-                fmt = IF_SVE_CM_3A;
+                assert(sopt == INS_SCALABLE_OPTS_NONE);
+                assert(isValidScalarDatasize(size));
+                fmt = IF_SVE_CO_3A;
             }
-            else if (insOptsScalableWithSimdScalar(opt))
+            else if (sopt == INS_SCALABLE_OPTS_WITH_SIMD_SCALAR)
             {
                 assert(isFloatReg(reg1));
                 assert(isValidVectorElemsize(size));
                 fmt = IF_SVE_CN_3A;
             }
             else
             {
-                assert(insOptsScalableWithScalar(opt));
-                assert(isGeneralRegister(reg1));
-                assert(isValidScalarDatasize(size));
-                fmt = IF_SVE_CO_3A;
+                assert(sopt == INS_SCALABLE_OPTS_NONE);
+                assert(isVectorRegister(reg1));
+                fmt = IF_SVE_CM_3A;
             }
             break;
 
         case INS_sve_cpy:
         case INS_sve_mov:
+            assert(insOptsScalable(opt));
             assert(isVectorRegister(reg1));
             assert(isLowPredicateRegister(reg2));
-            if (insOptsScalableWithSimdScalar(opt))
+            if (isGeneralRegisterOrSP(reg3))
             {
-                assert(isVectorRegister(reg3));
-                fmt = IF_SVE_CP_3A;
+                assert(sopt == INS_SCALABLE_OPTS_NONE);
+                fmt  = IF_SVE_CQ_3A;
+                reg3 = encodingSPtoZR(reg3);
             }
             else
             {
-                assert(insOptsScalableWithScalar(opt));
-                assert(isGeneralRegisterOrSP(reg3));
-                fmt  = IF_SVE_CQ_3A;
-                reg3 = encodingSPtoZR(reg3);
+                assert(sopt == INS_SCALABLE_OPTS_WITH_SIMD_SCALAR);
+                assert(isVectorRegister(reg3));
+                fmt = IF_SVE_CP_3A;
             }
+
             // MOV is an alias for CPY, and is always the preferred disassembly.
             ins = INS_sve_mov;
             break;
 
         case INS_sve_lasta:
         case INS_sve_lastb:
+            assert(insOptsScalable(opt));
             assert(isLowPredicateRegister(reg2));
             assert(isVectorRegister(reg3));
-            if (insOptsScalableWithSimdScalar(opt))
-            {
-                assert(isVectorRegister(reg1));
-                fmt = IF_SVE_CR_3A;
-            }
-            else
+            if (isGeneralRegister(reg1))
             {
-                assert(insOptsScalableWithScalar(opt));
+                assert(sopt == INS_SCALABLE_OPTS_NONE);
```

Comment:
nit in a follow-up PR: lets have a method for something like 
```c++
bool insOptsScalableOptsNone(sopt)
{
  return sopt == INS_SCALABLE_OPTS_NONE;
}
```

Reply:
Done as this was an easy find/replace.